### ISSUE TITLE
Container creation fails for versions 0.32.x and above

### ISSUE THREAD

**Issue by eshepelyuk** [crossplane-contrib/provider-aws#1531]
### What happened?
AWS provider controller container creation fails with error

```
 Warning  Failed     <invalid> (x4 over <invalid>)  kubelet            Error: failed to start container "provider-aws": Error response from daemon: OCI runtime create failed: container_linux.go:367: starting container process caused: chdir to cwd ("/home/nonroot") set in config.json failed: permission denied: unknown
```

### How can we reproduce it?
1. create EKS cluster
2. install Crossplane
3. Add AWS provider

### What environment did it happen in?
Crossplane version: 1.9.0, installed by Helm chart
k8s: EKS with k8s version 1.23

**Comment by pjanouse** [comment]
Release [v0.34.0](https://github.com/crossplane-contrib/provider-aws/releases/tag/v0.34.0) is still affected by this issue.

**Comment by pjanouse** [comment]
v0.36.0 still the same issue here.

**Comment by pjanouse** [comment]
@eshepelyuk was the fix released with the last published release `v0.40.0`? If yes, then I can report the issue is still here.

**Comment by pjanouse** [comment]
Issue isn't fixed thus raised a new one (https://github.com/crossplane-contrib/provider-aws/issues/1786) because I can't re-open this report.
